main.go

```diff
@@ -33,17 +39,28 @@ func main() {
 
 	r := prometheus.NewRegistry()
 	r.MustRegister(httpRequestsTotal)
+	r.MustRegister(httpRequestDuration)
 	r.MustRegister(version)
 
-	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	foundHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 		w.Write([]byte("Hello from example application."))
 	})
-	notfound := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	notfoundHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 	})
-	http.Handle("/", promhttp.InstrumentHandlerCounter(httpRequestsTotal, handler))
-	http.Handle("/err", promhttp.InstrumentHandlerCounter(httpRequestsTotal, notfound))
+
+	foundChain := promhttp.InstrumentHandlerDuration(
+		httpRequestDuration.MustCurryWith(prometheus.Labels{"handler": "found"}),
+		promhttp.InstrumentHandlerCounter(httpRequestsTotal, foundHandler),
+	)
+	notfoundChain := promhttp.InstrumentHandlerDuration(
```

Comment:
Is there a precedence for instrumenting the not found? Can you maybe look at some other cases where that was done. Thanks! If we can't find anything i would just leave it for the handler.